src/ray/util/event.cc

```diff
@@ -57,10 +57,22 @@ LogEventReporter::~LogEventReporter() { Flush(); }
 
 void LogEventReporter::Flush() { log_sink_->flush(); }
 
-std::string LogEventReporter::EventToString(const rpc::Event &event) {
-  std::stringstream result;
+std::string LogEventReporter::replaceLineFeed(std::string message) {
+  std::stringstream ss;
+  // If the message has \n or \r, we will replace with \\n
```

Comment:
Why do we need this? Is it for json serialization?

Reply:
Make sure the serialized json is one line in the event log. Agent will collect the event log line by line.

Reply:
Does this mean the format has tight coupling with the agent implementation here? 

I am asking it because we might want to just scrape the log file and store to some backend directly instead of reporting it to the dahsboard (not confirmed, but it is one of options). In this case, wouldn't this break the json format?

Reply:
Yep, a little hackly here. In our case, when we catch a backtrace to event, the event log written multiple lines. It is hard to parse. We just recover the message content in dashboard. Maybe we can optimize this in future? It is easy to remove this logic.